addons/mail/tests/test_ir_mail_server.py

```diff
@@ -398,46 +416,325 @@ def test_personal_mail_server(self):
+        self.assertEqual(mail_server_1.owner_limit_time, DATE_SEND_5)
+
+        # The CRON is late compared to the scheduled mails,
+        # it should re-schedule the mails, starting from the current time
+        # Should send in priority the old mails
+        DATE_SEND_6 = datetime(2025, 1, 1, 20, 25, 23)
+        with (
+            self.mock_smtplib_connection(),
+            self.mock_datetime_and_now(DATE_SEND_6),
+            self.assert_scheduled(mails_user_1, 5, 19, DATE_SEND_6),
+        ):
+            self.env['mail.mail'].process_email_queue()
+
+        self.assertEqual(mail_server_1.owner_limit_count, TEST_LIMIT)
+        self.assertEqual(mail_server_1.owner_limit_time, DATE_SEND_6)
+
+        # Finish sending the email
+        for i in range(2):
+            DATE_SEND_7 = datetime(2025, 1, 1, 20, 26 + i, 23)
+            with self.mock_smtplib_connection(), self.mock_datetime_and_now(DATE_SEND_7):
+                self.env['mail.mail'].process_email_queue()
+            self.assertEqual(mail_server_1.owner_limit_count, TEST_LIMIT)
+            self.assertEqual(mail_server_1.owner_limit_time, DATE_SEND_7)
+
+        DATE_SEND_8 = datetime(2025, 1, 1, 20, 28, 23)
+        with self.mock_smtplib_connection(), self.mock_datetime_and_now(DATE_SEND_8):
+            self.env['mail.mail'].process_email_queue()
+        self.assertEqual(mail_server_1.owner_limit_count, 4)
+        self.assertEqual(mail_server_1.owner_limit_time, DATE_SEND_8)
+
+        # We send 4 emails this minute, check that will send 1 and schedule the remaining
+        new_mails_user_1 = self.env["mail.mail"].with_user(user_1).sudo().create([
+            {'state': 'outgoing', 'email_to': 'target@test.com', 'email_from': user_1.email}
+            for i in range(TEST_LIMIT)
+        ])
+        DATE_SEND_9 = datetime(2025, 1, 1, 20, 28, 23)
+        with (
+            self.mock_smtplib_connection(),
+            self.mock_datetime_and_now(DATE_SEND_9),
+            self.assert_scheduled(new_mails_user_1, 1, len(new_mails_user_1), DATE_SEND_9),
+        ):
+            self.env['mail.mail'].process_email_queue()
+        self.assertEqual(mail_server_1.owner_limit_count, TEST_LIMIT)
+        self.assertEqual(mail_server_1.owner_limit_time, DATE_SEND_9)
+
+    @mute_logger('odoo.models.unlink')
+    def test_personal_mail_server_limit_many_recipient(self):
+        # Test with many recipients (should split the mail)
+        TEST_LIMIT = 5
+        self.env['ir.config_parameter'].set_param('mail.server.personal.limit.minutes', str(TEST_LIMIT))
+
+        partners = self.env['res.partner'].create([
+            {'name': f'Partner {i}', 'email': f'partner_{i}@test.com'}
+            for i in range(16)
+        ])
+        with self.mock_datetime_and_now("2025-01-01 20:30:23"):
+            mail = self.env["mail.mail"].with_user(self.user_employee).sudo().create({
+                'state': 'outgoing',
+                'email_to': 'target@test.com',
```

Comment:
Just to test we correctly keep formatting of email fields ^^
```suggestion
                'email_cc': '"Named Cc1" <cc.1@test.com>, "Named Cc2" <cc.2@test.com>',
                'email_to': '"Named To1" <to.1@test.com>, "Named To2" <to.1@test.com>',
```